src/v/crypto/key.cc

```diff
@@ -70,9 +68,11 @@ key::impl::impl(
   _private, internal::EVP_PKEY_ptr pkey, is_private_key_t is_private_key)
   : _pkey(std::move(pkey))
   , _is_private_key(is_private_key) {
+    static const absl::flat_hash_set<int> supported_key_types{EVP_PKEY_RSA};
+
     auto key_type = EVP_PKEY_get_base_id(_pkey.get());
 
-    if (!internal::supported_key_types.contains(key_type)) {
+    if (!supported_key_types.contains(key_type)) {
```

Comment:
It doesn't really need to be static, since it's not mutated.

Perhaps a constexpr `std::array` and `std::none_of` is sufficient given that it will be a small set:
```suggestion
    constexpr auto supported_key_types = std::to_array({EVP_PKEY_RSA});

    auto key_type = EVP_PKEY_get_base_id(_pkey.get());

    if (std::none_of(
          supported_key_types.begin(),
          supported_key_types.end(),
          [key_type](int k) { return k == key_type; })) {
```
Or `static assert` that the array is sorted and `std::binary_search`, but the complexity probably outweighs the benefit of searching values that probably all exist on the same cache line.

```suggestion
    constexpr auto supported_key_types = std::to_array({EVP_PKEY_RSA});
    static_assert(
      std::is_sorted(supported_key_types.begin(), supported_key_types.end()),
      "supported_key_types must be sorted");

    auto key_type = EVP_PKEY_get_base_id(_pkey.get());

    if (!std::binary_search(
          supported_key_types.begin(), supported_key_types.end(), key_type)) {
 ```

Reply:
agreed. i think using static here probably introduces overhead so that multi-threaded initialization works correclty then you pay for that on every acces.